src/platform/backends/qemu/firewall_config.cpp

```diff
@@ -108,110 +119,112 @@ void delete_iptables_rule(const QString& table, const QStringList& chain_and_rul
 
     if (!exit_state.completed_successfully())
         throw std::runtime_error(
-            fmt::format("Failed to delete iptables rule for table {}: {}", table, process->read_all_standard_error()));
+            fmt::format("Failed to delete firewall rule for table {}: {}", table, process->read_all_standard_error()));
 }
 
-auto get_iptables_rules(const QString& table)
+auto get_firewall_rules(const QString& firewall, const QString& table)
 {
-    auto process = MP_PROCFACTORY.create_process(iptables, QStringList() << wait << dash_t << table << list_rules);
+    auto process = MP_PROCFACTORY.create_process(firewall, QStringList() << wait << dash_t << table << list_rules);
 
     auto exit_state = process->execute();
 
     if (!exit_state.completed_successfully())
         throw std::runtime_error(
-            fmt::format("Failed to get iptables list for table {}: {}", table, process->read_all_standard_error()));
+            fmt::format("Failed to get firewall list for table {}: {}", table,
+                        (exit_state.error) ? exit_state.failure_message() : process->read_all_standard_error()));
```

Comment:
Ok, I see what you mean.  My question is, how do we want to handle the case if there isn't anything in `stderr`?  Is having something like `with output:` and nothing printed after it acceptable or should that part of the message be conditional on having something actually in `stderr`?

Reply:
I think `with output:` and nothing after is ok. It isn't the best natural language expression but it is still correct, in that it shows that the output was empty. Hopefully users won't see it often anyway. 

I often go with this `format("... var: {}", val)` approach because having `val` separate from the the main sentence avoids potential localization issues with different grammar constructions ("with no output" might translate differently). All without having to introduce if/else branching and mostly duplicate string literals. The added complexity is not worth it if it's not messaging that is forefront in UX, in my view. But anyway, I'll leave it up to you.

Reply:
Ok, I don't disagree with you and will update this to include both potential outputs.

Reply:
Done